### ISSUE TITLE
Dont pass around `BuildContext` for making snackbars

### ISSUE THREAD

**Issue by JAicewizard** [svthalia/Reaxit#292]
### What
Use a `GlobalKey<ScaffoldMessengerState>` instead of passing around a `BuildContext`

### Why
`BuildContext` can expire, and although it is possible to avoid using an expired `BuildContext` we don't know if the scaffold is still alive at the time of calling a snackbar.

### How
[Provide (a) suggestion(s) for how we could approach this.](https://github.com/flutter/flutter/blob/master/examples/api/lib/material/scaffold/scaffold_messenger.of.1.dart)

**Comment by DeD1rk** [comment]
That solution actually requires passing down the globalkey all over the widget tree, or lookign it up using another context. So it doesn't really help when compared to looking up from context synchronously.

**Comment by JAicewizard** [comment]
Yeah but in most cases its just in the widget itself, when its available its probably better to do this as its also faster
